File: solutions/prolog/rna-transcription/2/rna_transcription.pl
rna_transcription(Dna, Rna) :-
    string_chars(Dna, L),
    maplist(transcribe, L, RnaL),
    string_chars(Rna, RnaL).

transcribe('G', 'C').
transcribe('C', 'G').
transcribe('T', 'A').
transcribe('A', 'U').

File: solutions/clojure/rna-transcription/1/src/rna_transcription.clj
(ns rna-transcription)

(def transcribe-nucleotide
  {\G \C,
   \C \G,
   \T \A,
   \A \U})

(defn to-rna [dna]
  (assert (every? transcribe-nucleotide dna))
  (apply str (map transcribe-nucleotide dna)))
